src/dev_server/index.tsx

```diff
@@ -84,43 +91,79 @@ const DevServer = props => {
     };
   }
 
-  const pingServer = socket => {
-    // Continue to ping the sleeper app server until it responds.
-    return new Promise(() => {
-      NetInfo.fetch().then(netInfo => {
-        const netInfoDetails = netInfo?.details;
-        if (!netInfoDetails || !('ipAddress' in netInfoDetails)) {
-          console.error('[Sleeper] Failed to determine local IP address.');
-          return;
-        }
+  const startSocket = async () => {
+    const netInfo = await NetInfo.fetch();
+    const netInfoDetails = netInfo?.details;
+    const ipAddress = netInfoDetails?.ipAddress;
 
-        const message: SocketMessage = {_ip: netInfoDetails.ipAddress};
-        const json = JSON.stringify(message);
+    if (!netInfoDetails || !('ipAddress' in netInfoDetails)) {
+      console.error('[Sleeper] Failed to determine local IP address.');
+      return stopSocket();
+    }
 
-        (async function ping() {
-          socket.send(json, undefined, undefined, config.remoteSocketPort, config.remoteIP);
-          setTimeout(ping, 5000);
-        })();
-      });
+    connection.current = TcpSocket.createConnection({
+      port: config.remoteSocketPort,
+      host: config.remoteIP,
+      localAddress: ipAddress,
+      reuseAddress: true,
+    }, () => {
+      // When we establish a connection, send the IP address to the server
+      const message: SocketMessage = { _ip: ipAddress };
```

Comment:
do we actually still need to send this IP?

Reply:
Not technically - but we do need to send some kind of initial packet to trigger some connection callbacks. I can investigate in the future if we can remove this.